doctr/__main__.py

```diff
@@ -89,8 +89,7 @@ def get_parser():
         public repositories for the user. This option is not recommended
         unless you are using a separate GitHub user for deploying.""")
     configure_parser.add_argument("--no-upload-key", action="store_false", default=True,
-        dest="upload_key", help="""Don't automatically upload the deploy key to GitHub. If you select this
-        option, you will not be prompted for your GitHub credentials. """)
+        dest="upload_key", help="""Don't automatically upload the deploy key to GitHub.""")
```

Comment:
How about we just don't allowed `--no-upload-key` with private repos? Trying to prompt for password if it's not found is going to lead to a mess, because we'll need to get the `login_kwargs` out of `check_repo_exists` then so we have them to pass to other functions down the line.  Seems to me that `--no-upload-key` exists for people who want to add doctr docs to a public repo that they don't have access to, whereas if they have access to a private repo, they can probably upload keys no problem.